main.hpp

```diff
@@ -7,20 +7,23 @@
 
 #include "proposer.hpp"
 #include "acceptor.hpp"
+#include "batcher.hpp"
 
 class paxos {
 public:
     [[noreturn]] paxos();
 private:
+    int batcherIndex;
     std::vector<std::thread> participants {};  // A place to put threads so they don't get freed
     std::mutex clientsMutex;
     std::vector<int> clientSockets {};
 
     void startServer();
+    void startBatchers();
     void startProposers();
     void startAcceptors();
     [[noreturn]] void readInput();
-    void broadcastToProposers(const std::string& payload);
+    void broadcastToBatchers(const std::string& payload);
```

Comment:
I would rename to "sendToBatchers" since we're only sending to one now
Broadcast implies there are many receivers